common/app/dev/DevAssetsController.scala

```diff
@@ -25,11 +25,17 @@ object DevAssetsController extends Controller with ExecutionContexts {
     }
 
     // In dev mode we must consistently reload the asset map, to account for changes to the map.
-    val asset = localAssets.lookup.assets().getOrElse(path, findDevAsset.lift(path).getOrElse {
-        throw AssetNotFoundException(path)
-      })
+    val resolvedPath = findDevAsset.lift(
+      localAssets.lookup
+        .assets()
+        .get(path)
+        .map(_.path)
+        .getOrElse(path)
```

Comment:
needs a comment like `if that fails, try to pass the original path to findDevAsset`